server/resp/src/main/java/org/infinispan/server/resp/commands/generic/TOUCH.java

```diff
@@ -29,11 +28,9 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
                                                       ChannelHandlerContext ctx,
                                                       List<byte[]> arguments) {
 
-      CompletableFuture<Long> totalTouchCount = CompletableFutures.sequence(
-            arguments.stream()
-                  .map(key -> (CompletableFuture<Boolean>) handler.cache().touch(key, true))
-                  .collect(Collectors.toList())
-      ).thenApply(result -> result.stream().filter(r -> r.booleanValue()).count());
+      CompletionStage<Long> totalTouchCount = CompletionStages.performSequentially(arguments.iterator(),
+            k -> handler.cache().touch(k, true), Collectors.toList())
+            .thenApply(result -> result.stream().filter(r -> r.booleanValue()).count());
```

Comment:
```suggestion
      CompletionStage<Long> totalTouchCount = CompletionStages.performSequentially(arguments.iterator(),
            k -> handler.cache().touch(k, true), Collectors.counting());
```

I also want to bring up are we really sure we want to pass true here? For example this can revive an entry that has been expired for 30 minutes if the reaper interval is set to 1 hour.